openedx/core/djangoapps/profile_images/images.py

```diff
@@ -0,0 +1,130 @@
+"""
+Image file manipulation functions related to profile images.
+"""
+from cStringIO import StringIO
+
+from django.conf import settings
+from django.core.files.base import ContentFile
+from PIL import Image
+
+from ..user_api.accounts.api import get_profile_image_storage
+
+
+class DevMsg(object):
+    """
+    Holder for pseudo-constants.
+    """
+    FILE_TOO_LARGE = 'Maximum file size exceeded.'
+    FILE_TOO_SMALL = 'Minimum file size not met.'
+    FILE_BAD_TYPE = 'Unsupported file type.'
+    FILE_BAD_EXT = 'File extension does not match data.'
+    FILE_BAD_MIMETYPE = 'Content-Type header does not match data.'
+
+
+class ImageValidationError(Exception):
+    """
+    Exception to use when the system rejects a user-supplied source image.
+    """
+    pass
+
+
+def validate_uploaded_image(uploaded_file):
+    """
+    Raises ImageFileRejected if the server should refuse to use this
+    uploaded file as the source image for a user's profile image.  Otherwise,
+    returns nothing.
+    """
+    # validation code by @pmitros,
+    # adapted from https://github.com/pmitros/ProfileXBlock
+    # TODO: investigate if PIL has builtin methods for this
+
+    image_types = {
+        'jpeg': {
+            'extension': [".jpeg", ".jpg"],
+            'mimetypes': ['image/jpeg', 'image/pjpeg'],
+            'magic': ["ffd8"]
+        },
+        'png': {
+            'extension': [".png"],
+            'mimetypes': ['image/png'],
+            'magic': ["89504e470d0a1a0a"]
+        },
+        'gif': {
+            'extension': [".gif"],
+            'mimetypes': ['image/gif'],
+            'magic': ["474946383961", "474946383761"]
+        }
+    }
+
+    if uploaded_file.size > settings.PROFILE_IMAGE_MAX_BYTES:
+        raise ImageValidationError(DevMsg.FILE_TOO_LARGE)
+    elif uploaded_file.size < settings.PROFILE_IMAGE_MIN_BYTES:
+        raise ImageValidationError(DevMsg.FILE_TOO_SMALL)
+
+    # check the file extension looks acceptable
+    filename = str(uploaded_file.name).lower()
```

Comment:
nice catch.  i believe that name comes from what the file was named on the client, and thus could be anything.  I'll use `unicode` instead.